build_tools/github_actions/lint_llvm_commit.sh

```diff
@@ -35,23 +35,41 @@ PATH_TO_STABLEHLO_ROOT="$1"
 PATH_TO_LLVM_VERSION_TXT="$PATH_TO_STABLEHLO_ROOT/build_tools/llvm_version.txt"
 PATH_TO_WORKSPACE="$PATH_TO_STABLEHLO_ROOT/WORKSPACE.bazel"
 
-LLVM_DIFF=$(sed -n '/LLVM_COMMIT = /p' $PATH_TO_WORKSPACE | sed 's/LLVM_COMMIT = //; s/\"//g' | diff $PATH_TO_LLVM_VERSION_TXT -)
-
-update_llvm_commit_and_sha256() {
+retrieve_llvm_commit() {
   echo "Retrieving LLVM Commit..."
   export LLVM_COMMIT="$(<$PATH_TO_LLVM_VERSION_TXT)"
   echo "LLVM_COMMIT: $LLVM_COMMIT"
+}
+
+calculate_llvm_commit_sha256() {
   echo "Calculating SHA256..."
-  export LLVM_SHA256="$(curl -sL https://github.com/llvm/llvm-project/archive/$LLVM_COMMIT.tar.gz | shasum -a 256 | sed 's/ //g; s/-//g')"
-  echo "LLVM_SHA256: $LLVM_SHA256"
+  if [[ $(curl -sL https://github.com/llvm/llvm-project/archive/$LLVM_COMMIT.tar.gz | tr '\0' '\n') != "404: Not Found" ]]; then
```

Comment:
Also consider the `--head` flag to get headers only. Don't need to download the file

Reply:
I've added the `-I` flag (equivalent to `--head`). I'm still checking for the status code "404" though.

Reply:
How about avoid the grep and have curl provide the code. Two thoughts:

1. I like having 404 in the if stmt so we know exactly what we're special casing at the moment of the branch.
2. Lets echo the http code.. it may provide some insights if this does fail in CI.

```
HTTP_CODE=$(curl -sIL https://github.com/llvm/llvm-project/archive/$LLVM_COMMIT.tar.gz -o /dev/null -w "%{http_code}")
echo "HTTP Code: $HTTP_CODE"

if [[ "$HTTP_CODE" == "404" ]]; then
  echo "LLVM_COMMIT: $LLVM_COMMIT in $PATH_TO_LLVM_VERSION_TXT not found."
  exit 1
fi
```

Reply:
Good call. Will do.